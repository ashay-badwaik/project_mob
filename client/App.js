/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { ApolloClient, InMemoryCache, ApolloProvider } from '@apollo/client';

import { Text, View } from 'react-native';

import HomeScreen from './components/homeScreen';
import StoryScreen from './components/storyScreen';
import AddStoryScreen from './components/addStoryScreen';
import ChangeProfilePic from './components/changeProfilePic';


const client = new ApolloClient({
  uri: 'http://localhost:4000/',
  cache: new InMemoryCache(),
})

const Stack = createNativeStackNavigator();

const App = () => {
  return (
    <ApolloProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator initialRouteName="Home">
          <Stack.Screen name="Home" component={HomeScreen} options={{headerShown: false}}/>
          <Stack.Screen name="Story" component={StoryScreen}/>
          <Stack.Screen name="AddStory" component={AddStoryScreen} />
          <Stack.Screen name="ChangeProfilePic" component={ChangeProfilePic} />
        </Stack.Navigator>
      </NavigationContainer>
    </ApolloProvider>
  );
}

export default App;project_mov
