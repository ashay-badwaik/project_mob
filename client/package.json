{
  "name": "client",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@apollo/client": "^3.4.7",
    "@react-navigation/native": "^6.0.1",
    "@react-navigation/native-stack": "^6.0.2",
    "graphql": "^15.5.1",
    "react": "17.0.1",
    "react-native": "^0.64.2",
    "react-native-image-crop-picker": "^0.36.2",
    "react-native-safe-area-context": "^3.2.0",
    "react-native-screens": "^3.5.0",
    "react-native-svg": "^9.13.6",
    "react-native-svg-transformer": "^0.14.3"
  },
  "devDependencies": {
    "@babel/core": "7.15.0",
    "@babel/runtime": "7.14.8",
    "@react-native-community/eslint-config": "2.0.0",
    "babel-jest": "26.6.3",
    "eslint": "7.14.0",
    "jest": "26.6.3",
    "metro-react-native-babel-preset": "0.64.0",
    "react-test-renderer": "17.0.1"
  },
  "jest": {
    "preset": "react-native"
  }
}
