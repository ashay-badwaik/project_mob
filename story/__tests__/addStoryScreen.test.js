import 'react-native';
import React from 'react';
import { MockedProvider } from "@apollo/client/testing";
import AddStoryScreen from '../components/addStoryScreen';
import ImagePicker from 'react-native-image-crop-picker';
import AsyncStorage from '@react-native-async-storage/async-storage';

// Note: test renderer must be required after react-native.
import renderer, { act } from 'react-test-renderer';

jest.useFakeTimers()

const mockedParams = {
  route: {
    params: {
      storiesAvailable: true
    }
  },
  navigation: {
    navigate: jest.fn()
  }
}

const tree = renderer.create(
  <AddStoryScreen {...mockedParams} />
)

it('snapshot', () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )

  expect(tree).toMatchSnapshot()
});

it('image btn', () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )

  const btn = tree.root.findByProps({ testID: 'image' }).props;
  btn.onPress();

  // console.log(ImagePicker.openPicker());
  expect(ImagePicker.openPicker).toHaveBeenCalledTimes(1)
  // const img = tree.root.findByProps({testId:'display'}).props;
  // expect(img).not.toBeNull()
})

it('Create story btn', async () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )

  const btn = tree.root.findByProps({ testID: 'btn' }).props;
  await btn.onPress();

  expect(AsyncStorage.setItem).toHaveBeenCalled();
  expect(mockedParams.navigation.navigate).toBeCalledWith(expect.objectContaining({ name: 'Home' }))
})

it('caption input', () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )

  const setContent = jest.fn();


  const input = tree.root.findByProps({ testID: 'input' }).props;
  // input.onChangeText('abcd');
  expect(input).not.toBeNull()
})

it('addImg', () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )
  const txt = tree.root.findByProps({ testID: 'addImg' });
  expect(txt).not.toBeNull()
})

it('imgDisplay', async () => {
  const tree = renderer.create(
    <AddStoryScreen {...mockedParams} />
  )
  const btn = tree.root.findByProps({ testID: 'image' }).props;
  await act(async () => {
    await btn.onPress();
  })
  // await ImagePicker.openPicker().then((image)=>{console.log(image)})
  const img = tree.root.findByProps({ testID: 'display' });
  expect(img).not.toBeNull()

})
