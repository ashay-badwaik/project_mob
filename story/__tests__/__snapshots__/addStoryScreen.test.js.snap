// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTScrollView>
  <View>
    <Text
      style={
        Object {
          "color": "#fdbb21",
          "fontSize": 20,
          "marginHorizontal": 30,
          "marginVertical": 15,
          "textAlign": "center",
        }
      }
    >
      Create your story
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "borderColor": "#fdbb21",
          "borderRadius": 20,
          "borderStyle": "dashed",
          "borderWidth": 2,
          "height": 200,
          "justifyContent": "center",
          "marginHorizontal": 30,
          "marginVertical": 15,
          "opacity": 1,
        }
      }
      testID="image"
    >
      <Text
        style={
          Object {
            "color": "#fdbb21",
            "fontSize": 50,
            "lineHeight": 50,
            "marginHorizontal": 30,
            "marginVertical": 15,
            "textAlign": "center",
          }
        }
        testID="addImg"
      >
        Tap here to add image
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Add caption"
      placeholderTextColor="#fdbb21"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#fdbb21",
          "borderRadius": 20,
          "borderWidth": 2,
          "color": "black",
          "height": 100,
          "justifyContent": "flex-start",
          "marginHorizontal": 30,
          "marginVertical": 15,
          "textAlignVertical": "top",
        }
      }
      testID="input"
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fdbb21",
          "borderRadius": 20,
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 30,
          "marginVertical": 15,
          "opacity": 1,
        }
      }
      testID="btn"
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 30,
            "lineHeight": 30,
          }
        }
      >
        Create Story
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
